shared-module/displayio/Display.c

```diff
@@ -269,35 +289,67 @@ void displayio_display_end_transaction(displayio_display_obj_t* self) {
     self->end_transaction(self->bus);
 }
 
-void displayio_display_set_region_to_update(displayio_display_obj_t* self, uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1) {
-    self->send(self->bus, true, &self->set_column_command, 1);
-    bool isCommand = self->data_as_commands;
-    if (self->single_byte_bounds) {
-        uint8_t data[2];
-        data[0] = x0 + self->colstart;
-        data[1] = x1 - 1 + self->colstart;
-        self->send(self->bus, isCommand, (uint8_t*) data, 2);
-    } else {
-        uint16_t data[2];
-        data[0] = __builtin_bswap16(x0 + self->colstart);
-        data[1] = __builtin_bswap16(x1 - 1 + self->colstart);
-        self->send(self->bus, isCommand, (uint8_t*) data, 4);
+void displayio_display_set_region_to_update(displayio_display_obj_t* self, displayio_area_t* area) {
+    uint16_t x1 = area->x1;
+    uint16_t x2 = area->x2;
+    uint16_t y1 = area->y1;
+    uint16_t y2 = area->y2;
+    // Collapse down the dimension where multiple pixels are in a byte.
+    if (self->colorspace.depth < 8) {
+        uint8_t pixels_per_byte = 8 / self->colorspace.depth;
+        if (self->colorspace.pixels_in_byte_share_row) {
+            x1 /= pixels_per_byte;
+            x2 /= pixels_per_byte;
+        } else {
+            y1 /= pixels_per_byte;
+            y2 /= pixels_per_byte;
+        }
+    }
+
+    // Set column.
+    uint8_t data[5];
+    data[0] = self->set_column_command;
+    uint8_t data_length = 1;
+    if (!self->data_as_commands) {
+        self->send(self->bus, true, data, 1);
+        data_length = 0;
     }
-    self->send(self->bus, true, &self->set_row_command, 1);
     if (self->single_byte_bounds) {
-        uint8_t data[2];
-        data[0] = y0 + self->rowstart;
-        data[1] = y1 - 1 + self->rowstart;
-        self->send(self->bus, isCommand, (uint8_t*) data, 2);
+        data[data_length] = x1 + self->colstart;
+        data[data_length + 1] = x2 - 1 + self->colstart;
+        data_length += 2;
     } else {
-        uint16_t data[2];
-        data[0] = __builtin_bswap16(y0  + self->rowstart);
-        data[1] = __builtin_bswap16(y1 - 1 + self->rowstart);
-        self->send(self->bus, isCommand, (uint8_t*) data, 4);
+        x1 += self->colstart;
+        x2 += self->colstart - 1;
+        data[data_length] = x1 >> 8;
```

Comment:
Could do `data_length++` in each of these four subscripts, which is kind of an idiom, and then you don't need `data_length +=4` at the end.